TWTValidation/JSON Validator/Schema Validators/TWTJSONSchemaArrayValidator.m

```diff
@@ -0,0 +1,155 @@
+//
+//  TWTJSONSchemaArrayValidator.m
+//  TWTValidation
+//
+//  Created by Jill Cohen on 1/15/15.
+//  Copyright (c) 2015 Two Toasters, LLC.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "TWTJSONSchemaArrayValidator.h"
+
+@implementation TWTJSONSchemaArrayValidator
+
+- (instancetype)initWithMaximumItemCount:(NSNumber *)maximumItemCount
+                        minimumItemCount:(NSNumber *)minimumItemCount
+                     requiresUniqueItems:(BOOL)requiresUniqueItems
+                           itemValidator:(TWTValidator *)itemValidator
+                   indexedItemValidators:(NSArray *)indexedItemValidators
+                additionalItemsValidator:(TWTValidator *)additionalItemsValidator
+{
+    self = [super init];
+    if (self) {
+        _maximumItemCount = maximumItemCount;
+        _minimumItemCount = minimumItemCount;
+        _requiresUniqueItems = requiresUniqueItems;
+        _itemValidator = itemValidator;
+        _indexedItemValidators = [indexedItemValidators copy];
+        _additionalItemsValidator = additionalItemsValidator;
+    }
+    return self;
+}
+
+
+- (NSUInteger)hash
+{
+    return [super hash] ^ self.maximumItemCount.hash ^ self.minimumItemCount.hash ^ self.requiresUniqueItems ^ self.itemValidator.hash ^
+    self.indexedItemValidators.hash ^ self.additionalItemsValidator.hash;
+}
+
+
+- (BOOL)isEqual:(id)object
+{
+    if (![super isEqual:object]) {
+        return NO;
+    } else if (self == object) {
+        return YES;
+    }
+
+    typeof(self) other = object;
+    return [other.maximumItemCount isEqual:self.maximumItemCount] && [other.minimumItemCount isEqual:self.minimumItemCount] &&
+    other.requiresUniqueItems == self.requiresUniqueItems && [other.itemValidator isEqual:self.itemValidator] &&
+    [other.indexedItemValidators isEqual:self.indexedItemValidators] && [other.additionalItemsValidator isEqual:self.additionalItemsValidator];
```

Comment:
If any of these are objects and `nil`, this will return NO. You have to be a little more clever, basically checking, e.g.,   `(other.itemValidator == self.itemValidator || [other.itemValidator isEqual:self.itemValidator])`.